src/utils/request.ts

```diff
@@ -129,3 +133,20 @@ export function getRequestURL(
   const path = getRequestPath(event);
   return new URL(path, `${protocol}://${host}`);
 }
+
+export function setOriginalUrlPath(event: H3Event, url: string) {
+  // eslint-disable-next-line @typescript-eslint/no-extra-semi
+  (event.node.req as any).originalUrlPath = url;
+}
+
+export function getOriginalUrlPath(event: H3Event) {
+  return (event.node.req as any).originalUrlPath as string;
```

Comment:
I think the original URL could be simply an internal prop for event `event._originalPath` since it is mainly for advanced/internal usecases.

Reply:
Will remove as I've created #435